logic/src/commonMain/kotlin/com/wire/kalium/logic/data/publicuser/PublicUserMapper.kt

```diff
@@ -76,4 +83,22 @@ class PublicUserMapperImpl(private val idMapper: IdMapper) : PublicUserMapper {
 
     override fun fromPublicUserToLocalNotificationMessageAuthor(author: OtherUser?) =
         LocalNotificationMessageAuthor(author?.name ?: "", null)
+
+    override fun fromDaoAvailabilityStatusToUser(status: UserAvailabilityStatusEntity?): UserAvailabilityStatus? =
+        when (status) {
+            UserAvailabilityStatusEntity.AVAILABLE -> UserAvailabilityStatus.AVAILABLE
+            UserAvailabilityStatusEntity.BUSY -> UserAvailabilityStatus.BUSY
+            UserAvailabilityStatusEntity.AWAY -> UserAvailabilityStatus.AVAILABLE
+            UserAvailabilityStatusEntity.NONE -> UserAvailabilityStatus.NONE
+            null -> null
+        }
+
+    override fun fromUserAvailabilityStatusToDao(status: UserAvailabilityStatus?): UserAvailabilityStatusEntity? =
+        when (status) {
+            UserAvailabilityStatus.AVAILABLE -> UserAvailabilityStatusEntity.AVAILABLE
+            UserAvailabilityStatus.BUSY -> UserAvailabilityStatusEntity.BUSY
+            UserAvailabilityStatus.AWAY -> UserAvailabilityStatusEntity.AWAY
+            UserAvailabilityStatus.NONE -> UserAvailabilityStatusEntity.NONE
+            null -> null
```

Comment:
do we ever want to care about work with the null ? wouldn't it be better to return UserAvailibStatus.NONE?

Reply:
after some rethinking i think you're right: as far as `NONE ` means "no status" it can be used as default